django/forms/widgets.py

```diff
@@ -573,6 +573,24 @@ def render_options(self, selected_choices):
                 output.append(self.render_option(selected_choices, option_value, option_label))
         return '\n'.join(output)
 
+    def option_has_empty_value(self, choice):
+        choice_value = choice[0]
+        if isinstance(choice_value, str):
+            return len(choice_value) == 0
+        return choice_value is None
+
+    def use_required_attribute(self, initial):
+        # If first item has a value, rendering required attribute causes HTML validation error.
+        # More info: https://github.com/validator/validator/issues/47#issuecomment-77534952
+        super_result = super(Select, self).use_required_attribute(initial)
+
+        # Do checks only for select itself, not for its children
+        if self.allow_multiple_selected or hasattr(self, 'renderer'):
+            return super_result
+        else:
+            first_choice = next(iter(self.choices), None)
+            return super_result and first_choice is not None and self.option_has_empty_value(first_choice)
```

Comment:
You are right! I assumed `render_option` was always converting falsy values to an empty string. It might be worth keeping the method in this case.